mempool/src/core_mempool/mempool.rs

```diff
@@ -104,6 +104,7 @@ impl Mempool {
         db_sequence_number: u64,
         balance: u64,
         timeline_state: TimelineState,
+        local: bool,
```

Comment:
instead of introducing new variable local can we just use timeline_state enum instead. For transactions from shared mempool value would be TimelineState::NonQualified

Reply:
Using TimelineState does sound better than a new var. Please re-review the updated PR.